map/search_api.cpp

```diff
@@ -222,6 +223,34 @@ bool SearchAPI::SearchInViewport(ViewportSearchParams const & params)
   return Search(p, false /* forceSearch */);
 }
 
+bool SearchAPI::SearchForDiscovery(DiscoverySearchParams const & params)
+{
+  CHECK(params.m_onResults, ());
+  CHECK(!params.m_query.empty(), ());
+  CHECK_GREATER(params.m_itemsCount, 0, ());
+
+  SearchParams p;
+  p.m_query = params.m_query;
+  p.m_inputLocale = "en";
+  p.m_viewport = params.m_viewport;
+  p.m_position = params.m_viewportCenter;
+  p.m_maxNumResults = params.m_itemsCount;
+  p.m_mode = search::Mode::Viewport;
```

Comment:
Note that this effectively cancels current viewport search, do you really want this?

Reply:
I definitely want to use Viewport mode, so I guess in this case it's ok to cancel the previous search.

Reply:
Okay, np. But note that it's not a big deal to add your own mode, that will work exactly as search in Viewport.